rewrite-yaml/src/test/java/org/openrewrite/yaml/HelmTemplateParsingTest.java

```diff
@@ -0,0 +1,323 @@
+              """
+          )
+        );
+    }
+
+    @Test
+    void preserveFormattingWithHelmTemplates() {
+        rewriteRun(
+          yaml(
+            """
+              metadata:
+                name:     {{ .Release.Name }}
+                namespace:    {{ .Values.namespace }}
+                labels:
+                  app:    {{ .Values.appName }}
+                  chart:    {{ .Chart.Name }}-{{ .Chart.Version }}
+              spec:
+                replicas:    {{ .Values.replicas }}
+              """
+          )
+        );
+    }
+
+    @Test
+    void helmTemplatesWithComments() {
+        rewriteRun(
+          yaml(
+            """
+              # This is a Helm chart
+              apiVersion: v1
+              kind: Service
+              metadata:
+                name: {{ .Release.Name }} # The release name
+                labels:
+                  # Standard labels
+                  app: {{ .Values.appName }}
+                  # Chart information
+                  chart: {{ .Chart.Name }}
+              spec:
+                # Service configuration
+                type: {{ .Values.service.type | default "ClusterIP" }} # Default to ClusterIP
+                ports:
+                  - port: {{ .Values.service.port }} # The service port
+              """
+          )
+        );
+    }
+
+    @Test
+    void multilineHelmTemplates() {
+        rewriteRun(
+          yaml(
+            """
+              data:
+                script.sh: |
+                  {{- if .Values.enableDebug }}
+                  #!/bin/bash
+                  set -x
+                  {{- end }}
+                  echo "Starting {{ .Values.appName }}"
+                  {{- range .Values.commands }}
+                  {{ . }}
+                  {{- end }}
+              """
+          )
+        );
+    }
+
+    @Test
+    void nestedHelmTemplates() {
+        rewriteRun(
+          yaml(
+            """
+              spec:
+                containers:
+                - name: {{ .Values.container.name }}
+                  image: {{ include "app.image" (dict "imageRoot" .Values.image "global" .Values.global) }}
+                  env:
+                    - name: CONFIG
+                      value: {{ include "app.config" (merge (dict "specific" .Values.specific) .) | quote }}
+              """
+          )
+        );
+    }
+
+    @Test
+    void helmTemplatesInMappingKeys() {
+        rewriteRun(
+          yaml(
+            """
+              myconfig:
+                {{ .Values.keyName }}: "some value"
+                {{ .Values.dynamicKey }}: {{ .Values.dynamicValue }}
+                staticKey: {{ .Values.someValue }}
+              annotations:
+                {{ .Values.annotationKey }}: "enabled"
+                "{{ .Values.quotedKey }}": 'quoted value'
+              labels:
+                {{ include "app.labelKey" . }}: {{ include "app.labelValue" . }}
+              """
+          )
+        );
+    }
```

Comment:
Maybe add one more test case with a more complex template function.

https://helm.sh/docs/chart_template_guide/function_list/#logic-and-flow-control-functions